File: Editorial/112_Path_Sum/TypeScript/1_Recursion.ts
function hasPathSum(root: TreeNode | null, sum: number): boolean {
    if (root === null) return false;
    sum -= root.val;
    if (root.left === null && root.right === null)
        // if reach a leaf
        return sum === 0;
    return hasPathSum(root.left, sum) || hasPathSum(root.right, sum);
}

File: Editorial/0101-0200/0112_Path_Sum/C/1_Recursion.c
bool hasPathSum(struct TreeNode* root, int sum) {
    if (!root) return false;
    sum -= root->val;
    if (!root->left && !root->right)  // if reach a leaf
        return sum == 0;
    return hasPathSum(root->left, sum) || hasPathSum(root->right, sum);
}
